File: boscof-cs50-2017-x-pset6-20170423T095251Z/mario.py
import cs50

#ask user for valid input
while True:
    print("Height:", end ="")
    n = cs50.get_int()
    if n<24 and n>0:
        break
    
for i in range(1,n+1):
    #add in the initial spaces
    print(" " * (n-i), end="")
    #add in the blocks for the first half of the pyramid
    print("#" * i, end="")
    #add in the gap in the middle of the pyramid
    print("  ", end="")
    #add in the blocks for the second half of the pyramid
    print("#" * i, end="")
    #move on to the next line of the pyramid
    print("")

    
        

File: mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void){
int n;
    //ask user for valid input
    do 
    {
    printf("Height:\n");
    n = get_int();
    }
    while (n>23||n<0);
    
    //prints pyramid according to input specifications
    for(int i=0; i<n; i++)
    {
        //adds in the intial spaces
        for (int j=0; j<n-i-1; j++)
            printf(" ");
        //adds in the blocks for the first half of the pyramid along with the spaces in the middle
        for (int j=0; j<i+1; j++)
            printf("#");
        printf("  ");
        //adds in the blocks for the second half of the pyramid
        for (int j=0; j<i+1; j++)
            printf("#");
        printf("\n");
    }
}